src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+	recentTimeStamps.forEach((timestamp, index) => {
+		if (index === 0) {
+			sum += (Date.now() - timestamp);
+		} else {
+			sum += (recentTimeStamps[index - 1] - timestamp);
+		}
+	});
+	return sum / (recentTimeStamps.length);
+}
+
+function filterFeeds (): Promise<Array<FeedsObject>> {
+	return performReadQuery({
+		$: 'SELECT * FROM feeds WHERE lastrequesttime < (extract(epoch from NOW()) * 1000 - mtbu)'
+	});
+}
+
+function getRoomSpecificNews (): Promise<Array<RoomSpecificNews>> {
+	return performReadQuery({
+		$: `SELECT rooms.id as roomid, rooms.name as roomname, articles.rawjson as article, articles.url as url 
+			From rooms, articles
+			WHERE articles.terms @@ plainto_tsquery(rooms.name)`
+	});
+}
+
+function getMostRecentNewsForRoom (roomSpecificNewsArticles: Array<RoomSpecificNews>): Array<RoomSpecificNews> {
+	const sortedNewsArticles = _.orderBy(roomSpecificNewsArticles, (thread) => {
+		return thread.article.published;
+	}, [ 'desc' ]);
+	return _.uniqBy(sortedNewsArticles, 'roomid');
+}
+
+function hasNewNewsArticle (articlePubDate: number, lastupdatetime: number): boolean {
+	return articlePubDate > lastupdatetime;
+}
+
+function insertIntoFeeds (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `INSERT INTO feeds (url) 
+			VALUES ( &{url} )`,
+		url
+	} ]);
+}
+
+function insertNewArticles (url: string, article: Article, contentForLexemes: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `INSERT INTO articles (
+			url, rawjson, terms
+		) VALUES (
+			&{url}, &{rawjson}, to_tsvector(&{contentForLexemes}) 
+		)`,
+		url,
+		rawjson: JSON.stringify(article),
+		contentForLexemes
+	} ]);
+}
+
+function parseNewsFeed (url: string): Promise<Article> {
+	return parseFeed(url);
+}
+
+function truncateArticlesTable (): Promise<any> {
+	return performReadQuery({
+		$: 'TRUNCATE TABLE articles'
+	});
+}
+
+function updateFeedsOnNewArticles (lastupdatetime: number, url: string, mtbu: number): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: `UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000,
+			lastupdatetime = &{lastupdatetime}, mtbu = &{mtbu} 
+			WHERE url=&{url}`,
+		lastupdatetime,
+		url,
+		mtbu
+	} ]);
+}
+
+function updateFeedsOnNoNewArtcles (url: string): Promise<Array<{ rowCount: number }>> {
+	return performWriteQuery([ {
+		$: 'UPDATE feeds SET lastrequesttime = extract(epoch from NOW()) * 1000 WHERE url=&{url}',
+		url
+	} ]);
+}
+
+export const newsAggregator = (): Promise<Array<any>> => {
+	// Check if the feeds table is not empty.
+	return performReadQuery({
+		$: 'SELECT url FROM feeds',
+	})
+	.then((results) => {
+		// results: [{url: _}, ...]
+		if (results.length === 0) {
+			winston.info('feeds table is empty');
+			// insert seed url if feeds table is empty
+			const seedUrl = 'http://bangalore.citizenmatters.in/rss';
```

Comment:
Don't do this. Just work with the feeds in the database.

For testing, insert feeds into the database either manually or from your test, never from the module.